pkg/util/common.go

```diff
@@ -1347,25 +1304,32 @@ func DeleteCustomResourceDefinition(apiExtensionClient *apiextensionsclient.Clie
 }
 
 // WaitUntilPodsAreReady will wait for the pods to be ready
-func WaitUntilPodsAreReady(clientset *kubernetes.Clientset, namespace string, labelSelector string, timeoutInSeconds int64) (bool, error) {
-	// timer starts the timer for timeoutInSeconds. If the task doesn't completed, return error
-	timeout := time.NewTimer(time.Duration(timeoutInSeconds) * time.Second)
-	// ticker starts and execute the task for every n intervals
-	ticker := time.NewTicker(10 * time.Second)
-	for {
-		select {
-		case <-timeout.C:
-			ticker.Stop()
-			return false, fmt.Errorf("[NS: %s | Label: %s] the pods weren't ready - timing out after %d seconds", namespace, labelSelector, timeoutInSeconds)
-		case <-ticker.C:
-			ready, err := IsPodReady(clientset, namespace, labelSelector)
-			if err != nil || ready {
-				timeout.Stop()
-				ticker.Stop()
-				return ready, err
-			}
+func WaitUntilPodsAreReady(clientset *kubernetes.Clientset, namespace string, labelSelector string, timeoutInSeconds int64) error {
+	numOfTimesToCheck := 10
+	var duration time.Duration
+	if int(timeoutInSeconds) < numOfTimesToCheck {
+		// if timeoutInSeconds is less than 10 seconds, ticker will tick every timeoutInSeconds for numOfTimesToCheck
+		duration = time.Duration(timeoutInSeconds) * time.Second
+	} else {
+		// otherwise ticker will tick every 10th of timeoutInSeconds for numOfTimesToCheck
+		duration = time.Duration(int(timeoutInSeconds)/numOfTimesToCheck) * time.Second
+	}
+	// log.Debugf("Ticker will tick every %v seconds for %v times", duration, numOfTimesToCheck)
+
+	ticker := time.NewTicker(duration)
```

Comment:
ticker needs to be stopped.  Should use defer ticker.Stop().

Reply:
Good catch, of course good practice to stop it at the end, but I'm not sure how `golang` handles unstopped tickers.